clients/src/test/java/org/apache/kafka/common/serialization/SerializationTest.java

```diff
@@ -107,6 +103,15 @@ public void stringSerdeShouldSupportDifferentEncodings() {
         }
     }
 
+    @Test
+    public void listSerdeShouldShouldRoundtripInput() {
+        List<Integer> testData = Arrays.asList(1, 2, 3);
```

Comment:
Is it sufficient for testing ListSerde?

Reply:
We should also test `null` and empty array IMHO.

Please, add new test methods for both cases.